kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/kms/aws/KubeAwsKmsCloudTestKmsFacade.java

```diff
@@ -6,93 +6,95 @@
-        @Override
-        DescribeKeyResponse read(String alias) {
-            return runAwsKmsCommand(DESCRIBE_KEY_RESPONSE_TYPE_REF, awsCmd, KMS, DESCRIBE_KEY, PARAM_KEY_ID, ALIAS_PREFIX + alias);
-        }
+    @Override
+    public void stopKms() {
+        awsKmsCloud.delete();
+    }
 
-        @Override
-        void create(String alias) {
-            var createKeyResponse = runAwsKmsCommand(CREATE_KEY_RESPONSE_TYPE_REF, awsCmd, KMS, CREATE);
-            kekKeyId = createKeyResponse.keyMetadata().keyId();
+    @NonNull
+    @Override
+    public URI getAwsUrl() {
+        return awsKmsCloud.getAwsKmsUrl();
+    }
 
-            runAwsKmsCommand(awsCmd, KMS, CREATE_ALIAS, PARAM_ALIAS_NAME, ALIAS_PREFIX + alias, PARAM_TARGET_KEY_ID, kekKeyId);
-        }
+    @Override
+    public String getRegion() {
+        return awsKmsCloud.getRegion();
+    }
 
-        @Override
-        void rotate(String alias) {
-            var key = read(alias);
-            runAwsKmsCommand(awsCmd, KMS, ROTATE, PARAM_KEY_ID, key.keyMetadata().keyId());
-        }
+    @Override
+    public final Config getKmsServiceConfig() {
+        return new Config(getAwsUrl(), new InlinePassword(getKroxyliciousAccessKey()), new InlinePassword(getKroxyliciousSecretKey()), getRegion(), null);
+    }
 
-        @Override
-        void delete(String alias) {
-            var key = read(alias);
-            var keyId = key.keyMetadata().keyId();
-            runAwsKmsCommand(SCHEDULE_KEY_DELETION_RESPONSE_TYPE_REF,
-                    awsCmd, KMS, SCHEDULE_KEY_DELETION, PARAM_KEY_ID, keyId, PARAM_PENDING_WINDOW_IN_DAYS, "7" /* Minimum allowed */);
+    @Override
+    public String getSecretKey() {
+        return awsKmsCloud.getSecretKey();
+    }
 
-            runAwsKmsCommand(awsCmd, KMS, DELETE_ALIAS, PARAM_ALIAS_NAME, ALIAS_PREFIX + alias);
-        }
+    @Override
+    public String getAccessKey() {
+        return awsKmsCloud.getAccessKey();
+    }
 
-        @Override
-        ExecResult runAwsKmsCommand(String... command) {
-            ExecResult execResult = Exec.exec(null, List.of(command), Duration.ofSeconds(20), true, false, null);
+    private String getKroxyliciousSecretKey() {
+        return awsKmsCloud.getKroxyliciousSecretKey();
+    }
 
-            if (!execResult.isSuccess()) {
-                throw new KubeClusterException("Failed to run AWS Kms: %s, exit code: %d, stderr: %s".formatted(List.of(command),
-                        execResult.returnCode(), execResult.err()));
-            }
+    private String getKroxyliciousAccessKey() {
+        return awsKmsCloud.getKroxyliciousAccessKey();
+    }
 
-            return execResult;
+    @Override
+    public TestKekManager getTestKekManager() {
+        return new AwsKmsCloudTestKekManager();
+    }
+
+    class AwsKmsCloudTestKekManager extends AwsKmsTestKmsFacade.AwsKmsTestKekManager {
+
+        @Override
+        protected void rotate(String alias) {
```

Comment:
Currently it is needed to `@Override` this method because `LocalStack` does not support key rotation. I needed to make some methods `protected` to allow this overriding, but this class should be removed once key rotation is supported by `LocalStack`

Reply:
one idea: it'd probably do no harm to try the rotate endpoint and if it fails with a 404, fallback to our approach.  That way we have an implementation that will work with cloud or localstack.

Reply:
Done. Thanks for that! I didn't think about the HTTP response code, but actually it is not 404, it is 501.